### ISSUE TITLE
Revisit `options.override`

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#126997]
Now that every editor input has a `editorId`, I think we should revisit the `options.override` thing in untyped inputs and rather make it top level. Because setting the editor id should not be a optional thing but required in most cases. I.e. something like: `editorService.openEditor({ editorId: 'someEditor', resource: <some URI>})`.

We can still allow to select to pick an editor somehow, but I feel that `override` should just move out of the options.

I find this especially strange because in the `IEditorInput#matches` method we now have to probe onto `options.override`.

**Comment by lramos15** [comment]
Would we want to remove this from `IEditorOptions` too? I guess the concern comes with we would have to abandon overrides for typed editor inputs in that case and instead would just rely on their editorId

**Comment by bpasero** [comment]
Yeah good point, we possibly can only do this when no-one is using typed inputs anymore for opening because otherwise you wouldn't be able to open a typed editor with another editor.

**Comment by lramos15** [comment]
Just trying to revive or close out currently open debt items. Looks like we have a few cases of typed editor inputs being used for opening still and we should see what needs to be done to enable this case or we decide it's not worth it as the editor lifecycle is clean enough?

https://github.com/microsoft/vscode/blob/1df1845c5b0de85989676893b16b212a5908bda1/src/vs/workbench/browser/parts/editor/editorCommands.ts#L603
https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/debug/browser/breakpointsView.ts#L166
https://github.com/microsoft/vscode/blob/146049f09c70cf6e6e573c1f5fdcafbb980cf907/src/vs/workbench/contrib/debug/browser/debugEditorActions.ts#L165
https://github.com/microsoft/vscode/blob/8aebc83324fd6d541a92c5bc525321c4d11079f5/src/vs/workbench/contrib/debug/common/debugModel.ts#L462
https://github.com/microsoft/vscode/blob/ec9fc48cafd47af4adcd3ce8f03b4f23dc4d9253/src/vs/workbench/contrib/extensions/browser/abstractRuntimeExtensionsEditor.ts#L489
https://github.com/microsoft/vscode/blob/ec9fc48cafd47af4adcd3ce8f03b4f23dc4d9253/src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts#L872
https://github.com/microsoft/vscode/blob/4636c22e55a7d4ac49740257d72d2de060f5809a/src/vs/workbench/contrib/extensions/electron-sandbox/extensionProfileService.ts#L61
https://github.com/microsoft/vscode/blob/62de8dc3f78b8bef1eb62deeb093164a8d76c254/src/vs/workbench/contrib/extensions/electron-sandbox/extensionsAutoProfiler.ts#L199
https://github.com/microsoft/vscode/blob/1d111bf898550f33a02c0dd43f01a26428dca19f/src/vs/workbench/contrib/interactive/browser/interactive.contribution.ts#L357
https://github.com/microsoft/vscode/blob/1d111bf898550f33a02c0dd43f01a26428dca19f/src/vs/workbench/contrib/interactive/browser/interactive.contribution.ts#L395
https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/output/browser/output.contribution.ts#L206
https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/output/browser/output.contribution.ts#L272
https://github.com/microsoft/vscode/blob/776bd3d6d02dac6877a47772ad2bc72c4dcf37b0/src/vs/workbench/contrib/performance/browser/performance.contribution.ts#L54
https://github.com/microsoft/vscode/blob/0e8fcebfbc9f5ecb1e7db0750fc3eb5a9a8cad69/src/vs/workbench/contrib/search/browser/anythingQuickAccess.ts#L964
https://github.com/microsoft/vscode/blob/8d956244a57c2386fb1e9da57dd7b0c5a9055c71/src/vs/workbench/contrib/terminal/browser/terminalEditorService.ts#L197
https://github.com/microsoft/vscode/blob/8d956244a57c2386fb1e9da57dd7b0c5a9055c71/src/vs/workbench/contrib/terminal/browser/terminalEditorService.ts#L326
https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/webviewPanel/browser/webviewWorkbenchService.ts#L231
https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/webviewPanel/browser/webviewWorkbenchService.ts#L248
https://github.com/microsoft/vscode/blob/204d0247033ab7d2f6d1949ae6fd6d87e46a91b3/src/vs/workbench/contrib/welcomeGettingStarted/browser/gettingStarted.contribution.ts#L77
https://github.com/microsoft/vscode/blob/204d0247033ab7d2f6d1949ae6fd6d87e46a91b3/src/vs/workbench/contrib/welcomeGettingStarted/browser/gettingStarted.contribution.ts#L84
https://github.com/microsoft/vscode/blob/204d0247033ab7d2f6d1949ae6fd6d87e46a91b3/src/vs/workbench/contrib/welcomeGettingStarted/browser/gettingStarted.contribution.ts#L86
https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/welcomePage/browser/welcomePage.ts#L105
https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/welcomePage/browser/welcomePage.ts#L153
https://github.com/microsoft/vscode/blob/afb3140beb3d60393cbb70d02270e91a613e113b/src/vs/workbench/contrib/welcomeWalkthrough/browser/editor/editorWalkThrough.ts#L45
https://github.com/microsoft/vscode/blob/bf1c9e93913714c85b0db6f2e2690b0bddbab7e4/src/vs/workbench/contrib/workspace/browser/workspace.contribution.ts#L669
https://github.com/microsoft/vscode/blob/6827a04f445c9f4434760cdd7817f0c47d1e9f45/src/vs/workbench/services/history/browser/historyService.ts#L1860
https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/services/preferences/browser/preferencesService.ts#L237
https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/services/preferences/browser/preferencesService.ts#L319
https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/services/preferences/browser/preferencesService.ts#L350
https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/services/preferences/browser/preferencesService.ts#L358

To say there's a lot of cases is an understatement. Some are easier to fix than others, such as ones which have the group and can just use `group.openEditor` or ones which we could probably have adopt the resolver just as exclusive editors for their specified id. The other benefit of this is these editors aren't benefiting from the locking which you implemented due to the dependency on the resolver service for the editor list. I still believe webviews will be the hardest to fix though as they use a group identifier number and a webview editor cannot currently be easily registered or represented as an untyped input
